src/server/notifications/StreamingHttpChannel2023/StreamingHttpMetadataWriter.ts

```diff
@@ -19,8 +19,8 @@ export class StreamingHttpMetadataWriter extends MetadataWriter {
   }
 
   public async handle(input: { response: HttpResponse; metadata: RepresentationMetadata }): Promise<void> {
-    const resourcePath = input.metadata.identifier.value.replace(this.baseUrl, '');
-    const receiveFrom = `${this.baseUrl}${this.pathPrefix}${resourcePath}`;
+    const encodedUrl = encodeURIComponent(input.metadata.identifier.value);
+    const receiveFrom = `${this.baseUrl}${this.pathPrefix}${encodedUrl}`;
```

Comment:
I missed this last time, but when generating URLs you want to use the `joinUrl` utility function instead of concatenating strings. This reduces the chance of errors, if someone didn't add a slash to the end of their base URL for example.

Also if `baseUrl` only gets used like this you could just store `joinUrl(baseUrl, pathPrefix)` in a variable in the constructor instead of storing both separately and combining every time.